assets/app/scripts/services/routes.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("RoutesService", function() {
+    var isPortNamed = function(port) {
+      return angular.isString(port);
+    };
+
+    var getServicePortForRoute = function(targetPort, service) {
+      return _.find(service.spec.ports, function(servicePort) {
+        if (isPortNamed(targetPort)) {
+          // When using a named port in the route target port, it refers to the service port.
+          return servicePort.name === targetPort;
+        }
+
+        // Otherwise it refers to the container port (the service target port).
+        return servicePort.targetPort === targetPort;
+      });
+    };
+
+    var addRouteTargetWarnings = function(route, service, warnings) {
+      // Has the service been deleted?
+      if (!service) {
+        warnings.push('Routes to service "' + route.spec.to.name + '", but service does not exist.');
+        return;
+      }
+
+      var targetPort = route.spec.port ? route.spec.port.targetPort : null;
+      if (!targetPort) {
+        if (service.spec.ports.length > 1) {
+          warnings.push('Route has no target port, but service "' + service.metadata.name + '" has multiple ports. ' +
+                       'The route will round robin traffic across all exposed ports on the service.');
+        }
+
+        // Nothing else to check.
+        return;
+      }
+
+      // Warn when service doesn't have a port that matches target port.
+      var servicePort = getServicePortForRoute(targetPort, service);
+      if (!servicePort) {
+        if (isPortNamed(targetPort)) {
+          warnings.push('Service "' + service.metadata.name + '" has no named port "' + targetPort + '" set as route target port.');
+        } else {
+          warnings.push('Service "' + service.metadata.name + '" has no target port ' + targetPort + ' set as route target port.');
```

Comment:
same, maybe `Route target port is set to 8080 but service 'foo' does not expose that port.`